text/0012-corner-smoothing.md

```diff
@@ -0,0 +1,266 @@
+
+To be clear, this feature is not intended to be standardized as it is designed now. Instead, the intent is to satisfy one very specific need of developers at present. If we were to pursue the standardization route, we would have to fudamentally rethink our appraoch.
+
+<!--
+- Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this?
+- If this is an API proposal, could this be done as a JavaScript module or a native Node.js add-on
+  instead? Does the proposed change make Electron code easier or harder to read, understand,
+  and maintain?
+-->
+
+## Prior art
+
+### Smooth Rounded Corners
+
+* [Figma: Corner smoothing for sqiurcles](https://help.figma.com/hc/en-us/articles/360050986854-Adjust-corner-radius-and-smoothing#h_01HE150KTYDMPMNZ86K0GSTV68)
+  * [Figma blog: *Desparately seeking squircles* by Daniel Furse](https://www.figma.com/blog/desperately-seeking-squircles/)
+* [SwiftUI: `RoundedCornerStyle.continuous`](https://developer.apple.com/documentation/swiftui/roundedcornerstyle/continuous)
+* CSS Specification Drafts
+  * [CSS Borders Level 4 Issue](https://github.com/w3c/csswg-drafts/issues/10653)
+  * [CSS Backgrounds Level 4 Issue](https://github.com/w3c/csswg-drafts/issues/6296)
+  * [CSS Shapes Level 2 Issue](https://github.com/w3c/csswg-drafts/issues/10993)
+
+### CSS
+
+Electron has dabbled in having its "own" CSS rules before. `-webkit-app-region` is a CSS rule that allows developers to define what regions in their application can drag the window. While implemented by Chromium before Electron existed, most developers know of the CSS rule because of Electron and have only used it here. This CSS rule has been [proposed for standardization](https://github.com/w3c/csswg-drafts/issues/7017) as part of the proposed Window Controls Overlay feature.
+
+<!--
+Discuss prior art, both the good and the bad, in relation to this proposal. A few examples of what
+this can include are:
+
+- Does this feature exist in other frameworks and what experience have their community had?
+- Does this feature exist as a userland implementation, and what can be learned from it?
+- Is this related to a change upstream in Chromium or Node.js?
+- Does this proposal help Electron further align with evolving web standards?
+
+This section is intended to encourage you as an author to think about the lessons from prior
+implementations to provide readers of your RFC with a fuller picture. If there is no prior art,
+that is fine - your ideas are interesting to us whether they are brand new or if it is an
+adaptation from other technologies.
+-->
+
+## Unresolved questions
+
+### Resolve Through RFC Process
+
+- Is fingerprinting a major concern of this feature?
```

Comment:
Ideally, it'd be nice if a document could opt-in to using this feature so there's less risk of abuse.